File: SoftUni - C# OOP/Interfaces and Abstraction/Interfaces and Abstraction - Exercise/06. Food Shortage/Citizen.cs
namespace FoodShortage
{
    public class Citizen : IId, IBirthdate, IBuyer
    {
        private string name;
        private int age;
        private string id;
        private string birthday;
        private int food = 0;
        public Citizen(string name, int age, string id, string birthday)
        {
            this.Name = name;
            this.Age = age;
            this.Id = id;
            this.Birthday = birthday;
        }

        public string Name
        {
            get
            {
                return name;
            }
            set
            {
                name = value;
            }
        }
        public int Age
        {
            get
            {
                return age;
            }
            set
            {
                age = value;
            }
        }
        public string Id
        {
            get
            {
                return id;
            }
            set
            {
                id = value;
            }
        }

        public string Birthday
        {
            get
            {
                return birthday;
            }
            set
            {
                birthday = value;
            }
        }

        public int Food => food;

        public void BuyFood()
        {
            this.food += 10;
        }
    }
}

File: SoftUni - Java OOP/Interfaces and Abstraction/Interfaces and Abstraction - Exercises/_04_Food_Shortage/Citizen.java
package _04_Food_Shortage;

public class Citizen implements Person, Birthable, Identifiable, Buyer {
    private String name;
    private int age;
    private String id;
    private String birthDate;
    private int food;
    public Citizen(String name, int age, String id, String birthDate) {
        setName(name);
        setAge(age);
        setId(id);
       setBirthDate(birthDate);
       this.food = 0;
    }
    @Override
    public String getName() {
        return this.name;
    }

    @Override
    public int getAge() {
        return this.age;
    }

    private void setName(String name) {
        this.name = name;
    }

    private void setAge(int age) {
        this.age = age;
    }

    public String getId() {
        return id;
    }

    private void setId(String id) {
        this.id = id;
    }

    public String getBirthDate() {
        return birthDate;
    }

    private void setBirthDate(String birthDate) {
        this.birthDate = birthDate;
    }

    @Override
    public void buyFood() {
        this.food = 10;
    }

    @Override
    public int getFood() {
        return this.food;
    }
}
